Dockerfile

```diff
@@ -26,24 +29,35 @@ RUN mkdir /app \
             libpq-dev \
         && rm -rf /var/lib/apt/lists/*
 
-COPY ./requirements.txt ./docker/requirements-extra.txt ./setup.py ./MANIFEST.in ./README.md ./app/
-COPY superset /app/superset
-
+# First, we just wanna install requirements, which will allow us to utilize the cache
+# in order to only build iff requirements change
+COPY ./requirements.txt ./docker/requirements-extra.txt ./app/
 RUN cd /app \
-        && pip install -r requirements.txt -r requirements-extra.txt \
-        && pip install -e .
+        && pip install -r requirements.txt -r requirements-extra.txt
```

Comment:
NIT: Let's add a comment about getting `-extra` out of the lean build, and add instructions on how to layer up the drivers people may want

Reply:
Ended up getting it out. Now there's a new target called `dev` which contains it. Other images will build `lean`